lndclient/basic_client.go

```diff
@@ -17,7 +17,7 @@ import (
 // "basic" as it uses a global macaroon (by default the admin macaroon) for the
 // entire connection, and falls back to expected defaults if the arguments
 // aren't provided.
-func NewBasicClient(lndHost, tlsPath, macDir, network string) (lnrpc.LightningClient, error) {
+func NewBasicClient(lndHost, tlsPath, macPath, network string) (lnrpc.LightningClient, error) {
```

Comment:
I think we can avoid changing the set of arguments too much by using a functional argument here. This path will also mean that callers won't need to update their set of arguments they want the read only macaroon (admin being the default for now, though we should revisit, least authority principle n all).

Reply:
switched it to take in a `readonly` boolean param.

Reply:
ok updated to use functional optional args.